packages/nx/src/project-graph/project-graph-builder.ts

```diff
@@ -72,29 +71,51 @@ export class ProjectGraphBuilder {
   }
 
   /**
-   * Adds a dependency from source project to target project
+   * Adds static dependency from source project to target project
    */
-  addImplicitDependency(
+  addStaticDependency(
     sourceProjectName: string,
     targetProjectName: string
   ): void {
-    if (sourceProjectName === targetProjectName) {
-      return;
-    }
-    if (!this.graph.nodes[sourceProjectName]) {
-      throw new Error(`Source project does not exist: ${sourceProjectName}`);
+    this.addDependency(
+      sourceProjectName,
+      targetProjectName,
+      DependencyType.static
+    );
+  }
+
+  /**
+   * Adds dynamic dependency from source project to target project
+   */
+  addDynamicDependency(
```

Comment:
Adding a dynamic dependency should have the option of a file associated with it.